site/docs/latest/functions/quickstart.md

```diff
@@ -1,29 +1,176 @@
+$ tar xvf apache-pulsar-2.0.0-incubating-functions-preview-bin.tar.gz
+$ cd apache-pulsar-2.0.0-incubating-functions-preview
+$ bin/pulsar standalone \
+  --advertised-address 127.0.0.1
+```
+
+When running Pulsar in standalone mode, the `sample` {% popover tenant %} and `ns1` {% popover namespace %} will be created automatically for you. That tenant and namespace will be used throughout this tutorial.
+
+## Run a Pulsar Function in local run mode
+
+To run a function in local mode, i.e. outside our Pulsar cluster:
+
+```bash
+$ bin/pulsar-admin functions localrun \
+  --jar examples/api-examples.jar \
+  --className org.apache.pulsar.functions.api.examples.ExclamationFunction \
+  --inputs persistent://sample/standalone/ns1/exclamation-input \
+  --output persistent://sample/standalone/ns1/exclamation-output \
+  --name exclamation
+```
+
+The JAR file containing the function (written in Java) is included with the binary distribution you downloaded above.
+
+We can use the [`pulsar-client`](../../reference/CliTools#pulsar-client) CLI tool to publish a message to the input topic:
+
+```bash
+$ bin/pulsar-client produce persistent://sample/standalone/ns1/exclamation-input \
+  --num-produce 1 \
+  --messages "Hello world"
+```
+
+Here's what happens next:
+
+* The `Hello world` message that we published to the input {% popover topic %} (`persistent://sample/standalone/ns1/exclamation-input`) will be passed to the exclamation function that we're now running on our laptop
+* The exclamation function will process the message (providing a result of `Hello world!`) and publish the result to the output topic (`persistent://sample/standalone/ns1/exclamation-output`).
+* Pulsar will durably store the message data published to the output topic in [Apache BookKeeper](https://bookkeeper.apache.org) until a {% popover consumer %} consumes and {% popover acknowledges %} the message
+
+To consume the message, we can use the same [`pulsar-client`](../../reference/CliTools#pulsar-client) tool that we used to publish the original message:
+
+```bash
+$ bin/pulsar-client consume persistent://sample/standalone/ns1/exclamation-output \
+  --subscription-name my-subscription \
+  --num-messages 1
+```
+
+In the output, you should see the following:
+
+```
+----- got message -----
+Hello world!
+```
+
+Success! As you can see, the message has been successfully processed by the exclamation function. To shut down the function, simply hit **Ctrl+C**.
+
+## Run a Pulsar Function in cluster mode
+
+[Local run mode](#run-a-pulsar-function-in-local-run-mode) is useful for development and experimentation, but if you wanted to use Pulsar Functions in a real Pulsar deployment, you'd want to run them in **cluster mode**. In cluster mode, Pulsar Functions run *inside* your Pulsar cluster and are managed using the same `pulsar-admin functions` interface that we've been using thus far.
```

Comment:
wording can be changed here. Ppl can actually launch topologies in localrun to be run on clusters like mesos/k8. Running inside Pulsar Cluster is just yet another conv mode.